java-util/src/main/java/io/druid/java/util/common/granularity/AllGranularity.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.java.util.common.granularity;
+
+import com.google.common.collect.ImmutableList;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+import org.joda.time.format.DateTimeFormatter;
+
+/**
+ * AllGranularty buckets everything into a single bucket
+ */
+public final class AllGranularity extends Granularity
+{
+  // These constants are from JodaUtils in druid-common.
+  // Creates circular dependency.
+  // Will be nice to move JodaUtils here sometime
+  public static final long MAX_INSTANT = Long.MAX_VALUE / 2;
+  public static final long MIN_INSTANT = Long.MIN_VALUE / 2;
+
+  private static final AllGranularity INSTANCE = new AllGranularity();
+
+  private AllGranularity() {}
+
+  public static AllGranularity getInstance()
+  {
+    return INSTANCE;
+  }
+
+  @Override
+  public DateTimeFormatter getFormatter(Formatter type)
+  {
+    throw new UnsupportedOperationException("This method should not be invoked for this granularity type");
+  }
+
+  @Override
+  public DateTime increment(DateTime time)
+  {
+    return new DateTime(MAX_INSTANT);
+  }
+
+  @Override
+  public DateTime decrement(DateTime time)
+  {
+    throw new UnsupportedOperationException("This method should not be invoked for this granularity type");
+  }
+
+  @Override
+  public DateTime truncate(DateTime time)
+  {
+    return new DateTime(MIN_INSTANT);
+  }
+
+  @Override
+  public DateTime toDate(String filePath, Formatter formatter)
+  {
+    throw new UnsupportedOperationException("This method should not be invoked for this granularity type");
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    return new byte[]{0x7f};
```

Comment:
I don't know. @cheddar is the person who first committed this in, maybe he knows.